EIPS/eip-1102.md

```diff
@@ -11,13 +11,13 @@ created: 2018-05-04
 
 ## Simple summary
 
-This proposal describes a new way for DOM environments to expose the web3 API that requires user approval.
+This proposal describes a new way for DOM environments to expose the web3 provider that requires user approval.
 
 ## Abstract
 
 MetaMask and most other tools that provide access to web3-enabled environments do so automatically and without user consent. This exposes users of such environments to fingerprinting attacks since untrusted websites can check for a `web3` object and reliably identify web3-enabled clients.
```

Comment:
Can we change this paragraph to something like:

"MetaMask and most other tools that provide access to Ethereum-enabled environments expose a provider API through a `web3` object automatically and without user consent. This exposes users of such environments to fingerprinting attacks since untrusted websites can check for a `web3` object and reliably identify Ethereum-enabled clients."

Reply:
Same comment as above, specifications shouldn't refer to things specific products that may not be around in the future when the specification is being read.  This means don't talk about MetaMask, instead talk about this specification and other specifications.

Reply:
While I agree with other suggestions to remove historical context, it's slightly necessary here to explain the motivation behind this proposal. There's no EIP that defined the currently-unsafe auto-injection of web3, so it's not possible to refer to another proposal for context. A light historical context is warranted here, though reference to MetaMask should be nixed.

Reply:
Perhaps we would be better off describing the problem we are trying to solve rather than how previous implementations failed to solve it?  The abstract doesn't seem like the right place to discuss historic failures, but there may be value to including it at the _end_ of the Rationale section.  Also, I think we can avoid mentioning specific implementations and instead phrase it something like:
> The previous generation of Ethereum provider APIs would follow a pattern of injecting a provider object into the DOM directly. This resulted in exposing users to fingerprinting attacks by ... thus it does not address our desire with this specification to protect the user's privacy.

The way I am approaching this EIP is "if I was designing a new system today with no history, how would I author the EIP.  Then if necessary we can add some notes (such as in rationale or history or backward compatibility section) that discuss previous generation of Ethereum provider APIs, but I feel like the majority of the EIP should stand on its own without any historic references/knowledge, which includes the abstract and opening paragraphs of Rationale.